polaris-pipeline/coordinator/Functions/DurableEntity/Entity/CaseDurableEntity.cs

```diff
@@ -154,7 +154,8 @@ public Task<bool> AllDocumentsFailed()
                      (
                          cmsDocument.Status != DocumentStatus.Indexed ||
                          cmsDocument.CmsVersionId != incomingDocument.VersionId ||
-                         cmsDocument.IsOcrProcessed != incomingDocument.IsOcrProcessed
+                         cmsDocument.IsOcrProcessed != incomingDocument.IsOcrProcessed ||
+                         cmsDocument.CmsDocType?.DocumentTypeId != incomingDocument.CmsDocType?.DocumentTypeId
```

Comment:
No test for this specifically, but the associated tracing refactoring in this PR allows us to see records of case refreshes and of this logic taking effect.